seadssite/static/js/graph.js

```diff
@@ -148,42 +160,61 @@ function make_picker(func, repeat) {
     };
 }
 
-function pick_daily() {
+function pick_daily(panels) {
     var date = $("#daily-date").data("DateTimePicker").getDate();
-
     var start = Math.floor(date / 1000);
     var end = start + DAY_SECONDS;
-    fetch_graph(create_url(start, end));
+    var gran = 0;
+    var urls = [];
+    for(var i = 0; i < panels.length; i++) {
+        urls[i] = create_url(start, end, gran, panels[i]);
+    }
+    fetch_aggregate(urls, generate_chart, true, panels);
 }
 
-function pick_range() {
+function pick_range(panels) {
     var startDate = $("#range-start").data("DateTimePicker").getDate();
     var endDate = $("#range-end").data("DateTimePicker").getDate();
 
     var start = Math.floor(startDate / 1000);
     var end = Math.floor(endDate / 1000);
-
-    fetch_graph(create_url(start, end));
+    var gran = 0;
+    var urls = [];
+    for(var i = 0; i < panels.length; i++) {
+        urls[i] = create_url(start, end, gran, panels[i]);
+    }
+    fetch_aggregate(urls, generate_chart, true, panels);
 }
 
-function pick_live() {
+function pick_live(panels) {
     var end = Math.floor(Date.now() / 1000);
     var start = end - HOUR_SECONDS;
-    fetch_graph(create_url(start, end));
+    var gran = 0;
+    var urls = [];
+    for(var i = 0; i < panels.length; i++) {
+        urls[i] = create_url(start, end, gran, panels[i]);
+    }
+    fetch_aggregate(urls, generate_chart, true, panels);
 }
 
```

Comment:
do some de-duping of these three functions, seems like they could all be one function with some parameters passed in